pkg/scope/scope.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scope
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/utils/openstack/clientconfig"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Scope struct {
+	ProviderClient     *gophercloud.ProviderClient
+	ProviderClientOpts *clientconfig.ClientOpts
+
+	Logger logr.Logger
+	Client client.Client
+}
+
+type Params struct {
+	Client             *client.Client
+	ProviderClient     *gophercloud.ProviderClient
+	ProviderClientOpts *clientconfig.ClientOpts
+	Logger             *logr.Logger
+}
```

Comment:
These are (almost) the same? Do we need a separate struct?

Reply:
I adopted this from CAPA, where they used `interfaces` with getter methods for the actual scope, so they needed another `ScopeParams` struct to pass in the parameters to the `NewScope()` method.

You are right, it does not really make sense in this context.